examples/reddit.py

```diff
@@ -21,54 +20,11 @@
 train_loader = NeighborLoader(data, input_nodes=data.train_mask,
                               num_neighbors=[25, 10], shuffle=True, **kwargs)
 
-subgraph_loader = NeighborLoader(copy.copy(data), input_nodes=None,
-                                 num_neighbors=[-1], shuffle=False, **kwargs)
-
-# No need to maintain these features during evaluation:
-del subgraph_loader.data.x, subgraph_loader.data.y
-# Add global node index information.
-subgraph_loader.data.num_nodes = data.num_nodes
-subgraph_loader.data.n_id = torch.arange(data.num_nodes)
-
-
-class SAGE(torch.nn.Module):
-    def __init__(self, in_channels, hidden_channels, out_channels):
-        super().__init__()
-        self.convs = torch.nn.ModuleList()
-        self.convs.append(SAGEConv(in_channels, hidden_channels))
-        self.convs.append(SAGEConv(hidden_channels, out_channels))
-
-    def forward(self, x, edge_index):
-        for i, conv in enumerate(self.convs):
-            x = conv(x, edge_index)
-            if i < len(self.convs) - 1:
-                x = x.relu_()
-                x = F.dropout(x, p=0.5, training=self.training)
-        return x
-
-    @torch.no_grad()
-    def inference(self, x_all, subgraph_loader):
-        pbar = tqdm(total=len(subgraph_loader.dataset) * len(self.convs))
-        pbar.set_description('Evaluating')
-
-        # Compute representations of nodes layer by layer, using *all*
-        # available edges. This leads to faster computation in contrast to
-        # immediately computing the final representations of each batch:
-        for i, conv in enumerate(self.convs):
-            xs = []
-            for batch in subgraph_loader:
-                x = x_all[batch.n_id.to(x_all.device)].to(device)
```

Comment:
I am still a bit confused about this. Here, in the previous example, we explicitly convert intermediate embeddings to CPU, while we are now storing them on GPU.

Reply:
Do you mean it may not be as fast as the previous example if we store intermediate embeddings to the CPU? Or do we need to maintain the semantics?

Reply:
Sorry for late reply. I mean that the previous example stored intermediate embeddings on CPU, and we can only maintain the same runtime if we now store them on GPU. I think we still need to look into what causes the real slowdown.